source/crud/tests/write/findOneAndReplace-upsert.yml

```diff
@@ -45,3 +47,45 @@ tests:
                     - {_id: 2, x: 22}
                     - {_id: 3, x: 33}
                     - {_id: 4, x: 44}
+    -
+        description: "FindOneAndReplace when no documents match with id specified with upsert returning the document before modification"
```

Comment:
The two tests "without id specified" above previously existed in `findOneAndReplace.yml` but these "with id specified" variants are new. This makes the test cases consistent with what we already had in `replaceOne.yml` for the `update` command.